src/bin/generate_test_vector.rs

```diff
@@ -0,0 +1,75 @@
+use color_eyre::eyre::{Result, WrapErr};
+use prio::{
+    field::{Field126, Field32, Field64, Field80, FieldElement, FieldPriov2},
+    test_vector::EnpaTestVector,
+};
+use serde::Serialize;
+use std::str::FromStr;
+use structopt::StructOpt;
+
+#[derive(Debug)]
+enum FieldArgument {
+    FieldPriov2,
+    Field32,
+    Field64,
+    Field80,
+    Field126,
+}
+
+impl FromStr for FieldArgument {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "fieldpriov2" => Ok(Self::FieldPriov2),
+            "field32" => Ok(Self::Field32),
+            "field64" => Ok(Self::Field64),
+            "field80" => Ok(Self::Field80),
+            "field126" => Ok(Self::Field126),
+            _ => Err("unknown field"),
+        }
+    }
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "generate-test-vector",
+    about = "Generate test vectors for ENPA",
+    rename_all = "kebab-case",
+    version = env!("CARGO_PKG_VERSION"),
+)]
+struct Options {
+    /// Dimension (number of bins) of the inputs
+    #[structopt(short, long, required = true)]
+    dimension: usize,
+    /// Number of inputs to generate
+    #[structopt(short, long, required = true)]
+    number_of_inputs: usize,
+    /// Field to use to encode inputs
+    #[structopt(short, long, required = true)]
+    field: FieldArgument,
+}
+
+fn generate_and_print_vector<F: FieldElement + Serialize>(options: Options) -> Result<()> {
+    let test_vector: EnpaTestVector<F> =
+        EnpaTestVector::new(options.dimension, options.number_of_inputs)
+            .wrap_err("failed to create test vector")?;
+    let json =
+        serde_json::to_string(&test_vector).wrap_err("failed to encode test vector to JSON")?;
+    println!("{}", json);
+
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+    let options = Options::from_args();
+
+    match options.field {
+        FieldArgument::FieldPriov2 => generate_and_print_vector::<FieldPriov2>(options),
+        FieldArgument::Field32 => generate_and_print_vector::<Field32>(options),
+        FieldArgument::Field64 => generate_and_print_vector::<Field64>(options),
+        FieldArgument::Field80 => generate_and_print_vector::<Field80>(options),
+        FieldArgument::Field126 => generate_and_print_vector::<Field126>(options),
```

Comment:
Test vectors are mighty big. Is there a need to generate them for each field?

Reply:
I only generated a `FieldPriov2` vector, but I wanted to support more types. However, this tool only supports emitting test vectors in the Priov2 specific encoding and encryption scheme, which will only ever use `FieldPriov2`, so I can simplify all this code by removing the generic `FieldElement` parameter.

Reply:
I see, in theory would could use other fields for ENPA. Do you know if there are plans to do so? If this isn't planned, then I would hard-code FieldPriov2.